awx/ui_next/src/screens/WorkflowApproval/WorkflowApprovalList/WorkflowApprovalList.jsx

```diff
@@ -0,0 +1,277 @@
+import React, { useCallback, useEffect } from 'react';
+import { useLocation, useRouteMatch } from 'react-router-dom';
+import { withI18n } from '@lingui/react';
+import { t } from '@lingui/macro';
+import { Card, PageSection } from '@patternfly/react-core';
+import { WorkflowApprovalsAPI } from '../../../api';
+import PaginatedDataList, {
+  ToolbarDeleteButton,
+} from '../../../components/PaginatedDataList';
+import AlertModal from '../../../components/AlertModal';
+import ErrorDetail from '../../../components/ErrorDetail';
+import DataListToolbar from '../../../components/DataListToolbar';
+import WorkflowApprovalListItem from './WorkflowApprovalListItem';
+import useRequest, {
+  useDeleteItems,
+  useDismissableError,
+} from '../../../util/useRequest';
+import useSelected from '../../../util/useSelected';
+import { getQSConfig, parseQueryString } from '../../../util/qs';
+import useWsWorkflowApprovals from './useWsWorkflowApprovals';
+import WorkflowApprovalListApproveButton from './WorkflowApprovalListApproveButton';
+import WorkflowApprovalListDenyButton from './WorkflowApprovalListDenyButton';
+
+const QS_CONFIG = getQSConfig('workflow_approvals', {
+  page: 1,
+  page_size: 20,
+  order_by: '-started',
+});
+
+function WorkflowApprovalsList({ i18n }) {
+  const location = useLocation();
+  const match = useRouteMatch();
+
+  const {
+    result: { results, count, relatedSearchableKeys, searchableKeys },
+    error: contentError,
+    isLoading: isWorkflowApprovalsLoading,
+    request: fetchWorkflowApprovals,
+  } = useRequest(
+    useCallback(async () => {
+      const params = parseQueryString(QS_CONFIG, location.search);
+      const [response, actionsResponse] = await Promise.all([
+        WorkflowApprovalsAPI.read(params),
+        WorkflowApprovalsAPI.readOptions(),
+      ]);
+      return {
+        results: response.data.results,
+        count: response.data.count,
+        relatedSearchableKeys: (
+          actionsResponse?.data?.related_search_fields || []
+        ).map(val => val.slice(0, -8)),
+        searchableKeys: Object.keys(
+          actionsResponse.data.actions?.GET || {}
+        ).filter(key => actionsResponse.data.actions?.GET[key].filterable),
+      };
+    }, [location]),
+    {
+      results: [],
+      count: 0,
+      relatedSearchableKeys: [],
+      searchableKeys: [],
+    }
+  );
+
+  useEffect(() => {
+    fetchWorkflowApprovals();
+  }, [fetchWorkflowApprovals]);
+
+  // TODO: update QS_CONFIG to be safe for deps array
```

Comment:
Is this a note for your current work that you forgot to remove, or is this future work?

Reply:
that's an old TODO of mine, copied over from somewhere else I think